### ISSUE TITLE
Script fails with IndexError: list index out of range

### ISSUE THREAD

**Issue by GeertJohan** [vrothberg/vgrep#5]
```
$ vgrep main
Traceback (most recent call last):
  File "/home/geertjohan/bin/vgrep", line 246, in <module>
    main()
  File "/home/geertjohan/bin/vgrep", line 104, in main
    slocs.append(SLOC(hit))
  File "/home/geertjohan/bin/vgrep", line 31, in __init__
    self.file = SLOC.parse_file(string)
  File "/home/geertjohan/bin/vgrep", line 38, in parse_file
    return RE_FILE.findall(string)[0]
IndexError: list index out of range
```

**Comment by vrothberg** [comment]
Thanks for reporting :-)

I don't find a way to get the IndexError.  Which version of grep / git are you using?

**Comment by GeertJohan** [comment]
```
$ git version
git version 2.1.4
$ grep --version
grep (GNU grep) 2.20
```

**Comment by GeertJohan** [comment]
vgrep repo at `009b725eba18a4382e796de840a544d208059937`

**Comment by vrothberg** [comment]
Thanks again!

I still cannot retrace what's wrong :(  I put a try-catch around the constructor so that it avoids vgrep to fail:
   https://github.com/vrothberg/vgrep/commit/cede06ee60486eae49d9e4c1cabf50d6fe6eebc7

**Comment by GeertJohan** [comment]
Sure no problem. Thanks for making vgrep!

It's still not working on my machine though. I hope the following helps:

```
$ mkdir vgrep-test && cd vgrep-test
$ echo "foobar1" > file1.txt
$ echo "foobar2" > file2.txt
$ git init
Initialized empty Git repository in /home/geertjohan/vgrep-test/.git/
$ git add .
$ vgrep foobar
Could not parse 'file1.txt:1:foobar1'
list index out of range
Could not parse 'file2.txt:1:foobar2'
list index out of range
Index  Source File  Source Line  Content

```

**Comment by GeertJohan** [comment]
I can't reproduce this error in a debian wheezy container, but I did find an other error.

`$HOME/.cache` folder wasn't created. Maybe check for this and create when missing? ping @gregkh 

$ docker run -ti buildpack-deps:wheezy-scm /bin/bash
root@3985bad053da:/# export PS1="$ "
$ git version
git version 1.7.10.4
$ grep --version
grep (GNU grep) 2.12
$ git clone https://github.com/vrothberg/vgrep.git
Cloning into 'vgrep'...
remote: Counting objects: 102, done.
remote: Compressing objects: 100% (89/89), done.
remote: Total 102 (delta 57), reused 56 (delta 11), pack-reused 0
Receiving objects: 100% (102/102), 167.70 KiB, done.
Resolving deltas: 100% (57/57), done.
$ cd vgrep/
$ ./vgrep vgrep
Traceback (most recent call last):
  File "./vgrep", line 256, in <module>
    main()
  File "./vgrep", line 113, in main
    dump(slocs)
  File "./vgrep", line 244, in dump
    pickle.dump(data, open(CACHE, "wb"))
IOError: [Errno 2] No such file or directory: '/root/.cache/vgrep'
$ mkdir /root/.cache
$ ./vgrep vgrep
Index  Source File  Source Line  Content

0      Makefile     1            TARGET = vgrep
1      Makefile     5            install: vgrep
2      README.md    1            # vgrep
3      README.md    13           directory by calling **vgrep FOO**.  You can also specify multiple arguments
4      README.md    14           to search for.  vgrep prints the occurrences in the format "Index  Source File
5      README.md    16           in an editor.  vgrep caches the results of the last call, so that you can
6      README.md    17           re-open your previous results by simply calling **vgrep** without arguments.
7      README.md    21           ![](https://github.com/vrothberg/vgrep/blob/master/screenshots/grep_example.png)
8      README.md    23           Note that vgrep pipes the output to 'less' for outputs bigger than 100 indexes.
9      README.md    3            **vgrep** is a Python script to search strings in a given source tree.  It is
10     README.md    31           To visit a specific location pass **'--show INDEX'**.  vgrep will then open the
11     README.md    33           vgrep defaults to *vim* if the *EDITOR* environment variable is not set.
12     README.md    37           [~/linux/kernel/irq]$ vgrep --show 2
13     README.md    40           ![](https://github.com/vrothberg/vgrep/blob/master/screenshots/show_example.png)
14     vgrep        19           CACHE = HOMEDIR + "/.cache/vgrep"
15     vgrep        4            Visit https://github.com/vrothberg/vgrep for more information."""
```

**Comment by vrothberg** [comment]
Thanks for your help!

**Could not parse 'file1.txt:1:foobar1'**

... now that's interesting.  It seems like that the problem is in the regex module.  The regular expression works on all machines that I have access to.  I try to figure out what's wrong.

**Comment by gregkh** [comment]
Yes, I'll fix up the .cache/ directory thing, sorry about that.

**Comment by GeertJohan** [comment]
when I redirect the output of `git grep -In vgrep` in the vgrep repo to a file, and compare the filesizes on the docker wheezy and my local machine, there is a difference. Local machine output is 1564 bytes whereas docker wheezy output is 1148 bytes.
Could it be color color coding --??

Yes, found it, it is color coding. Creating PR shortly.

**Comment by gregkh** [comment]
mkdir ~/.cache/ should now be addressed with pull request #11
